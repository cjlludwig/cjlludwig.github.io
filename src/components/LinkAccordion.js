import { useState } from "react";
import Accordion from 'react-bootstrap/Accordion';
import { ReactTinyLink } from 'react-tiny-link';
import linkPic from '../static/images/web-link.png';


import 'bootstrap/dist/css/bootstrap.min.css';

function LinkAccordion({ links=[] }) {
  const [ openKey, setOpen ] = useState(null);
  // const proxy = process.env.REACT_APP_ENV === "dev" ? "https://proxy.cors.sh" : "";

  return (
    <Accordion >
    {
      links.map((data, index) => {
        const { name, url, description } = data;
        return (
          <Accordion.Item key={index} eventKey={index + 1}>
            <Accordion.Header
              onClick={() => setOpen(index)}
            >
             {name}
            </Accordion.Header>
            <Accordion.Body>
              {openKey === index &&
                <ReactTinyLink
                  width="100%"
                  description={description}
                  defaultMedia={linkPic}
                  cardSize="small"
                  showGraphic={true}
                  maxLine={3}
                  minLine={2}
                  url={url}
                  proxyUrl={""}
                  requestHeaders={ { 
                    "Access-Control-Allow-Origin": "*" ,
                    "Access-Control-Allow-Headers":"*",
                    'Access-Control-Allow-Methods': 'HEAD, GET, OPTIONS',
                    "Access-Control-Max-Age": 1000,
                  } }
                />
              } 
            </Accordion.Body>
          </Accordion.Item>
        )
      })
    }
  </Accordion>
  );
}

export default LinkAccordion;
