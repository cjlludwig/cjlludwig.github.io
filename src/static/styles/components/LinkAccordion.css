.link-pic {
  height: 20vmin;
  padding: 10px;
}

.link-span {
  display: inline-flex;
  width: 100%;
  align-items: center;
}

.link-card {
  width: 100%;
}