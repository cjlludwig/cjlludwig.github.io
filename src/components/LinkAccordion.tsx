import { useState } from "react";
import Accordion from 'react-bootstrap/Accordion';
import { Links } from '../models/links';
import linkPic from '../static/images/web-link.png';
import { Card } from "react-bootstrap";
import '../static/styles/components/LinkAccordion.css';
import 'bootstrap/dist/css/bootstrap.min.css';

function LinkAccordion<Props extends { links: Links[] }>(props: Props) {
  const { links=[] } = props;
  const [ openKey, setOpen ] = useState<number | null>(null);

  return (
    <Accordion >
    {
      links.map((data, index) => {
        const { name, url, description } = data;
        return (
          <Accordion.Item key={index} eventKey={`${index+1}`}>
            <Accordion.Header
              onClick={() => setOpen(index)}
            >
             {name}
            </Accordion.Header>
            <Accordion.Body>
              {openKey === index &&
              <a href={url} style={{ textDecoration: "none", color: "inherit" }}>
                <span className="link-span">
                  <img src={linkPic} className="link-pic" alt="default link" />
                  <Card className="link-card">
                    <Card.Header>{name}</Card.Header>
                    <Card.Body>{description}</Card.Body>
                    <Card.Footer style={{ textDecoration: "underline", color: "blue" }}>{url}</Card.Footer>
                  </Card>
                </span>
              </a>
              } 
            </Accordion.Body>
          </Accordion.Item>
        )
      })
    }
  </Accordion>
  );
}

export default LinkAccordion;
