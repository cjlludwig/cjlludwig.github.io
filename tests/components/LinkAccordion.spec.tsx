import React from "react";
import renderer from "react-test-renderer";
// import { render, fireEvent, waitFor } from "@testing-library/react";

import LinkAccordion from "../../src/components/LinkAccordion";

// function renderLinkAccordion(props: Partial<Props> = {}) {
//   const defaultProps: Props = {
//     links: []
//   };

//   return render(<LinkAccordion {...defaultProps} {...props} />);
// }

describe("<LinkAccordion />", () => {
  test("renders correctly", () => {
    const tree = renderer
      .create(<LinkAccordion links={[]} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  //   const { findByTestId } = renderLinkAccordion();

  //   const LinkAccordion = await findByTestId("link-accordion");
  
  //   expect(LinkAccordion).toHaveFormValues({
  //     username: "",
  //     password: "",
  //     remember: true
  //   });
  //   expect(true).toEqual(true);
});