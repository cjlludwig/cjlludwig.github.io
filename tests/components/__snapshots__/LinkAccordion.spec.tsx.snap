// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinkAccordion /> renders correctly 1`] = `
<div
  className="accordion"
  data-testid="link-accordion"
/>
`;
